packages/pyright-internal/src/common/pathUtils.ts

```diff
@@ -92,19 +92,21 @@ export function forEachAncestorDirectory(
 
 export function getDirectoryPath(pathString: string): string {
     if (isUri(pathString)) {
-        return Utils.dirname(URI.parse(pathString).with({ fragment: '' })).toString();
+        return Utils.dirname(URI.parse(pathString).with({ fragment: '', query: '' })).toString();
     }
     return pathString.substr(0, Math.max(getRootLength(pathString), pathString.lastIndexOf(path.sep)));
 }
 
+const _schemePattern = /^\w[\w\d+.-]*$/;
```

Comment:
If this is used only in one function, define it within that function. If it's used within multiple functions, move it to the top of the file and give it a better name so it's clear what it means within the global scope of this source file.

Reply:
It's only used in this function but I didn't want it to recreate the regex every time the function was called. I'll move it to the top.